test/container.test.js

```diff
@@ -0,0 +1,99 @@
+  // add in the id3 content
+  .concat(Array.apply(null, Array(35)).map(() => 0x00));
+
+const id3DataWithFooter = []
+  // id3 header is 20 bytes with footer
+  // "we have a footer" is the sixth byte
+  // 10th byte is length of 0x23 or 35 in decimal
+  // so a total length of 55
+  .concat(stringToBytes('ID3').concat([0x00, 0x00, 0xFF, 0x00, 0x00, 0x00, 0x23]))
+  // add in the id3 content
+  .concat(Array.apply(null, Array(45)).map(() => 0x00));
+
+const testData = {
+  'webm': [0x1A, 0x45, 0xDf, 0xA3],
+  'flac': stringToBytes('fLaC'),
+  'ogg': stringToBytes('OggS'),
+  'aac': [0xFF, 0xF1],
+  'mp3': [0xFF, 0xFB],
+  '3gp': [0x00, 0x00, 0x00, 0x00].concat(stringToBytes('ftyp3g')),
+  'mp4': [0x00, 0x00, 0x00, 0x00].concat(stringToBytes('ftypiso')),
+  'ts': [0x47]
+};
+
+QUnit.module('detectContainerForBytes');
+
+QUnit.test('should identify known types', function(assert) {
+  Object.keys(testData).forEach(function(key) {
+    const data = new Uint8Array(testData[key]);
+
+    assert.equal(detectContainerForBytes(testData[key]), key, `found ${key} with Array`);
+    assert.equal(detectContainerForBytes(data.buffer), key, `found ${key} with ArrayBuffer`);
+    assert.equal(detectContainerForBytes(data), key, `found ${key} with Uint8Array`);
+  });
+
+  const mp4Bytes = new Uint8Array([0x00, 0x00, 0x00, 0x00].concat(stringToBytes('styp')));
+
+  assert.equal(detectContainerForBytes(mp4Bytes), 'mp4', 'styp mp4 detected as mp4');
+
+  // mp3 and aac audio can have id3 data before the
+  // signature for the file, so we need to handle that.
+  ['mp3', 'aac'].forEach(function(type) {
+    const dataWithId3 = new Uint8Array([].concat(id3Data).concat(testData[type]));
+    const dataWithId3Footer = new Uint8Array([].concat(id3DataWithFooter).concat(testData[type]));
+
+    assert.equal(detectContainerForBytes(dataWithId3), type, `id3 skipped and ${type} detected`);
+    assert.equal(detectContainerForBytes(dataWithId3Footer), type, `id3 + footer skipped and ${type} detected`);
+  });
+
+  const notTs = [testData.ts].concat(fillerArray(188));
+  const longTs = [testData.ts]
+    .concat(fillerArray(187))
+    .concat(testData.ts);
+
+  assert.equal(detectContainerForBytes(longTs), 'ts', 'long ts data is detected');
+  assert.equal(detectContainerForBytes(notTs), '', 'ts missing 0x47 at 188 is not ts at all');
+  assert.equal(detectContainerForBytes(otherMp4Data), 'mp4', 'fmp4 detected as mp4');
+  assert.equal(detectContainerForBytes(new Uint8Array()), '', 'no type');
+});
+
+const createBox = function(type) {
+  const size = 0x20;
+
+  // size bytes
+  return [0x00, 0x00, 0x00, size]
+    // box identfier styp
+    .concat(stringToBytes(type))
+    // filler data for size minus identfier and size bytes
+    .concat(fillerArray(size - 8));
+};
+
+QUnit.module('isLikelyFmp4');
+QUnit.test('works as expected', function(assert) {
+  const fmp4Data = []
+    .concat(createBox('styp'))
```

Comment:
Since `styp` is optional, probably worth adding a test without it too.

Reply:
added